src/elements/content-sidebar/activity-feed/comment/Replies.js

```diff
@@ -0,0 +1,160 @@
+        onSuccess: ?Function,
+        onError: ?Function,
+    ) => void,
+    onReplySelect?: (isSelected: boolean) => void,
+    onShowReplies?: () => void,
+    parentID: string,
+    replies?: CommentType[],
+    repliesTotalCount?: number,
+};
+
+type Props = ReplyProps & {
+    currentUser?: User,
+    getAvatarUrl: GetAvatarUrlCallback,
+    getMentionWithQuery?: Function,
+    getUserProfileUrl?: GetProfileUrlCallback,
+    mentionSelectorContacts?: SelectorItems<>,
+    translations?: Translations,
+};
+
+const Replies = ({
+    currentUser,
+    getAvatarUrl,
+    getMentionWithQuery,
+    getUserProfileUrl,
+    hasReplies = false,
+    isAlwaysExpanded = false,
+    isRepliesLoading,
+    mentionSelectorContacts,
+    onHideReplies = noop,
+    onReplyDelete = noop,
+    onReplyEdit = noop,
+    onReplySelect = noop,
+    onShowReplies = noop,
+    parentID,
+    replies = [],
+    repliesTotalCount = 0,
+    translations,
+}: Props) => {
+    const { length: repliesLength } = replies;
+    const repliesToLoadCount = Math.max(repliesTotalCount - repliesLength, 0);
+
+    const onHideRepliesHandler = () => {
+        if (repliesLength) {
+            onHideReplies(replies[repliesLength - 1]);
+        }
+    };
+
+    const renderButton = () => {
+        if (isAlwaysExpanded || isRepliesLoading) {
+            return null;
+        }
+
+        if (repliesTotalCount > repliesLength) {
+            return (
+                <PlainButton
+                    className="bcs-Replies-toggle"
+                    onClick={onShowReplies}
+                    type="button"
+                    data-testid="show-replies-button"
+                >
+                    <FormattedMessage values={{ repliesToLoadCount }} {...messages.showReplies} />
+                </PlainButton>
+            );
+        }
+        if (repliesTotalCount > 1 && repliesTotalCount === repliesLength) {
+            return (
+                <PlainButton
+                    className="bcs-Replies-toggle"
+                    onClick={onHideRepliesHandler}
+                    type="button"
+                    data-testid="hide-replies-button"
+                >
+                    <FormattedMessage {...messages.hideReplies} />
+                </PlainButton>
+            );
+        }
+
+        return null;
+    };
+
+    const getReplyPermissions = (reply: CommentType): BoxCommentPermission => {
+        const { permissions: { can_delete = false, can_edit = false, can_resolve = false } = {} } = reply;
+        return {
+            can_delete,
+            can_edit,
+            can_resolve,
+        };
+    };
+
+    if (!hasReplies) {
+        return null;
+    }
+
+    return (
+        <div className="bcs-Replies">
+            {renderButton()}
+            <ul className="bcs-Replies-list">
+                {isRepliesLoading && (
+                    <div className="bcs-Replies-loading" data-testid="replies-loading">
```

Comment:
I don't think `<ul><div></div></ul>` is valid markup. Is this structure correct?

Reply:
Good point, will modify